ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -398,34 +399,40 @@ public Query visitQuery(final SqlBaseParser.QueryContext context) {
 
       final boolean pullQuery = context.EMIT() == null && !buildingPersistentQuery;
 
-      final Optional<ResultMaterialization> resultMaterialization;
+      final Optional<OutputRefinement> outputRefinement;
 
       if (pullQuery) {
-        resultMaterialization = Optional.empty();
+        outputRefinement = Optional.empty();
       } else if (buildingPersistentQuery) {
-        resultMaterialization = Optional.of(Optional
+        outputRefinement = Optional.of(Optional
             .ofNullable(context.resultMaterialization())
             .map(rm -> rm.FINAL() == null
-                ? ResultMaterialization.CHANGES
-                : ResultMaterialization.FINAL
+                ? OutputRefinement.CHANGES
+                : OutputRefinement.FINAL
             )
-            .orElse(ResultMaterialization.CHANGES));
+            .orElse(OutputRefinement.CHANGES));
         // Else must be a push query, which must specify a materialization
       } else {
-        resultMaterialization = Optional
+        outputRefinement = Optional
             .of(context.resultMaterialization().CHANGES() == null
-                ? ResultMaterialization.FINAL
-                : ResultMaterialization.CHANGES
+                ? OutputRefinement.FINAL
+                : OutputRefinement.CHANGES
             );
       }
+      final Optional<RefinementInfo> refinementInfo;
+      if (!pullQuery) {
+        refinementInfo = Optional.of(RefinementInfo.of(outputRefinement));
+      } else {
+        refinementInfo = Optional.empty();
+
+      }
```

Comment:
```suggestion
      final Optional<RefinementInfo> refinementInfo = outputRefinement.map(RefinementInfo::of);
```
This way we don't duplicate the `if (!pullQuery)` logic from above and just rely on knowing that if it's a pull query the `outputRefinement` will already be `Optional.empy()`